protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/detector/XmpDetector.java

```diff
@@ -0,0 +1,254 @@
+/*
+* OCA CONTRIBUTION ACKNOWLEDGEMENT - NOT PART OF LEGAL BOILERPLATE
+* DO NOT DUPLICATE THIS COMMENT BLOCK WHEN CREATING NEW FILES!
+*
+* This file was contributed to the OpenNMS(R) project under the
+* terms of the OpenNMS Contributor Agreement (OCA).  For details on
+* the OCA, see http://www.opennms.org/index.php/Contributor_Agreement
+*
+* Contributed under the terms of the OCA by:
+*
+* Bobby Krupczak <rdk@krupczak.org>
+* THE KRUPCZAK ORGANIZATION, LLC
+* http://www.krupczak.org/
+*/
+
+/**
+ * OpenNMS XMP Detector allows for discovery of service/protocols via 
+ * provisiond.  Our detector is pretty simple right now.  All it does is
+ * attempt to establish an XmpSession with a system and if it succeeds,
+ * it queries a few core MIB variables and returns success.
+ * Future enhancements could include more capability information like
+ * supported MIBs, modules/plugins, and various dependencies to
+ * determine if something is a server for some service (noted by an
+ * inbound dependency relationship).
+ */
+
+package org.opennms.protocols.xmp.detector;
+
+import java.net.InetAddress;
+import java.util.Date;
+import org.opennms.netmgt.provision.support.BasicDetector;
+import org.opennms.netmgt.provision.support.AbstractDetector;
+import org.opennms.netmgt.provision.ServiceDetector;
+import org.opennms.netmgt.provision.SyncServiceDetector;
+import org.krupczak.xmp.*;
+import org.opennms.netmgt.config.xmpConfig.XmpConfig;    
+import org.opennms.netmgt.protocols.xmp.config.XmpConfigFactory;
+import org.opennms.netmgt.protocols.xmp.collector.XmpCollector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component
+/**
+ * XmpDetector class
+ * @author rdk <rdk@krupczak.org>
+ * @version $Id: $
+ *
+ */
+@Scope("prototype")
+public class XmpDetector implements SyncServiceDetector
+{
+    // class variables
+
+    private static final String DEFAULT_SERVICE_NAME = "XMP";
+    private static final Logger LOG = LoggerFactory.getLogger(XmpDetector.class);
+    private static int XMP_DEFAULT_TIMEOUT = 3000;
+    private static int XMP_DEFAULT_RETRIES = 1;
+    
+    // instance variables
+    SocketOpts sockopts;
+    XmpConfig protoConfig;
+    int xmpPort;
+    int xmpTimeout;
+    int xmpRetries;
+    String xmpAuthenUser;
+    String xmpServiceName; 
+    String m_ipMatch;
+    Date createTimeDate;
+
+    /* 
+     * @param serviceName a {@link java.lang.String} object
+     * @param port an int specifying tcp port number
+     */
+    
+    public XmpDetector(String serviceName, int port)
+    {
+	//super(serviceName,port,XMP_DEFAULT_TIMEOUT,XMP_DEFAULT_RETRIES);
+	
+        // set default config
+        xmpPort = port;
+        xmpAuthenUser = new String("xmpUser"); 
```

Comment:
Please create a static member for the default authenticated user. 
Please do not create a String using `new String("...")` Simply use `"..."`.